integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/V2CatalogParser.java

```diff
@@ -0,0 +1,25 @@
+package io.openlineage.spark3.agent.lifecycle.plan;
+
+import io.openlineage.spark.agent.util.PlanUtils;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.sql.connector.catalog.CatalogPlugin;
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions;
+import org.apache.spark.sql.execution.datasources.v2.jdbc.JDBCTableCatalog;
+
+public class V2CatalogParser {
+  public static Path getCatalogLocation(CatalogPlugin catalog) {
+    //     TODO: add more catalogs (iceberg?)
+    if (catalog instanceof JDBCTableCatalog) {
+      return parseJdbc((JDBCTableCatalog) catalog);
+    }
```

Comment:
I think this is a great example of what's missing in the `PlanUtils` handling of the `CatalogTable`. Rather than add this Catalog parser, can we pass the `CatalogTable` (which we can extract with `command.catalog().loadTable(command.tableName())` ) to the `PlanUtils` method and add some extra handling there for JDBC URLs?

Reply:
Sometimes, we cannot do `loadTable(command.tableName())` since the table does not exist yet - or we can, but only on complete event.

In this case, also, we have `Table` interface introduced in Spark 3, so we need to have something like `PlanUtils3` for this particular case.